libosmscout/src/osmscout/navigation/PositionAgent.cpp

```diff
@@ -118,6 +118,7 @@ namespace osmscout {
       double nearest=std::numeric_limits<double>::max();
 
       PositionAgent::Position position;
+      position.coord.Set(coord.GetLat(), coord.GetLon());
```

Comment:
Isn't this just `position.coord=coord`? A direct assign of coord object is safer, because it avoid lat/lon getting passed inverted.

Possibly PositionAgent should get an explicit constructor to enforce passing a valid coord?

Reply:
You are right, the operator=() is preferable